src/generateApis.ts

```diff
@@ -91,15 +91,14 @@ export const ${serviceName} = async (
     },
     ${queryParamsTypeName ? "queryParams" : "undefined"},
     ${requestBody ? "requestBody" : "undefined"},
-    overrideConfig({
-        headers: {
-          "Content-Type": "${contentType}",
-          Accept: "${accept}",
-          ${headerParams ? "...headerParams," : ""}
-        },
+    {
+      headers: {
+        "Content-Type": "${contentType}",
+        Accept: "${accept}",
+        ${headerParams ? "...headerParams," : ""}
       },
-      configOverride,
-    )
+      ...configOverride,
+    },
   ))
```

Comment:
Do not remove overrideConfig from here. We don't want to lose headers added here.
If you remove overrideConfig, and user add a property to headers, headers will be replaced with new headers.

Reply:
I'm not sure that that is true, if you see [Axios configuration](https://github.com/axios/axios/blob/master/README.md#config-order-of-precedence), then axios merges configuration in the order:
1) library defaults
2) instance defaults
3) whatever you define in the actual requests.

Reply:
Yes, axios merge, not replace, but spread object do not merge nested object, in this case headers.

For example, if I call a method like this, I lose some property like content-type, because you spread that over headers.
```js
getUser({headers:{language:'en'}})
```

Reply:
ah, yes, you're right.. it's late..
I'll rework things in the pr..

Reply:
What about using https://www.npmjs.com/package/lodash.merge (or something like that) to do deep merge of config/overrideConfig objects?

Reply:
Ended up re-adding overrideConfig method (just moved to the services file, as it's only used there). 
I think that lodash.merge could be an option for a future PR though, so we do a propper merge of objects, instead of only headers.